### PR TITLE
[fix](pipeline) failure if memory is not enough

### PR DESCRIPTION
**Pull request by dataroaring**
## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by dataroaring** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

File: be/src/pipeline/pipeline_task.cpp [hunk 1]
```diff
@@ -302,7 +302,7 @@ Status PipelineTask::execute(bool* eos) {
     }
     // The status must be runnable
     if (!_opened && !_fragment_context->is_canceled()) {
-        RETURN_IF_ERROR(_open());
```
Comment (Gabriel39) [inline be/src/pipeline/pipeline_task.cpp:305]: Exception is already caught by caller

File: be/src/pipeline/pipeline_fragment_context.cpp [hunk 2]
```diff
@@ -337,13 +337,13 @@ Status PipelineFragmentContext::prepare(const doris::TPipelineFragmentParams& re
     for (PipelinePtr& pipeline : _pipelines) {
         SCOPED_TIMER(_prepare_all_pipelines_timer);
         pipeline->children().clear();
-        RETURN_IF_ERROR(pipeline->prepare(_runtime_state.get()));
+        RETURN_IF_ERROR_OR_CATCH_EXCEPTION(pipeline->prepare(_runtime_state.get()));
```
Comment (Gabriel39) [inline be/src/pipeline/pipeline_fragment_context.cpp:340]: Maybe we should catch exception by `prepare`'s caller

File: be/src/vec/exprs/vexpr.cpp [hunk 3]
```diff
@@ -343,7 +343,7 @@ Status VExpr::create_tree_from_thrift(const std::vector<TExprNode>& nodes, int*
     // create root expr
     int root_children = nodes[*node_idx].num_children;
     VExprSPtr root;
-    RETURN_IF_ERROR(create_expr(nodes[*node_idx], root));
```
Comment (Gabriel39) [inline be/src/vec/exprs/vexpr.cpp:346]: Catch exception in `FoldConstantExecutor::fold_constant_vexpr`

**Pull request closed without merging**